### ISSUE TITLE
Replace translated code of MPI_Get_count for its Bamboo Function counterpart

### ISSUE THREAD

**Issue by SergioMartin86** [SergioMartin86/bamboo6#38]
Right now MPI_Get_count is being translated manually by Bamboo. I reckon this is not necessary, and a function call can be used instead.

Also review the preprocess function in MPI_Processing.C file.

**Comment by scottb5612** [comment]
Actually, I took out GetCount() some time ago.
It wasn't supported by Bamboo last fall, and I came up with a cleaner solution that
solve some other issues. Basically, I compute what are called communication
schedules, which are then used in the inspector-executor model. See genRecvLists.c and genRecvLists_no_mpi.c.

The original code sized the receive buffers so that they were sufficiently large to hold
any incoming message holding "ghost cell" mesh data (though cart3d's meshes are unstructured,
by the analogy applies).  Using the receiveLIst routines, each core knows how
much ghost data it will be receiving from  each communicating neighbor. So it can size the buffer exactly.
My optimizations aren't complete: each time a routine collects ghost cells, it sizes the buffer.
But in this application, the mesh is static. I haven't gotten around to factoring out the receive
list generation.

There was also another impact on the logic of ghost cell communication.
The old code used Wait_any in  a loop to process each neighbor's data.
With the receive lists, I can just use Wait_all. Bamboo didn't support Wait_any due to 
tarragon related issues.

In Tan's words (Nov 7, 2014)
Tan: (Nov 7)
I need some more time to think about MPI_ANY_SOURCE, MPI_ANY_TAG, and
MPI_Waitany. This is not an implementation problem, but instead an
execution model problem. In our framework, a task can't deny a message
injection from the runtime. Thus, the task has to passively receive the
injected message that it doesn't need at the current time and stores the
message somewhere without knowing how it will process the message. The
only way that the task tracks such messages later is via source and tag.
Thus, I still have not figured out what we can do if an MPI_ANY_XXX is
used.

I think what Tan was saying was that the put/inject mechanism of
Tarragon doesn’t enable the destination to reason about message tags,
nor communicators  nor message sources, though this was not followed up.
